tests/test_optimizer/test_dist_lamb.py

```diff
@@ -0,0 +1,180 @@
+        self.fc1 = Linear1D_Col.from_native_module(fc1, process_group=tp_group, gather_output=False, overlap=True)
+        self.fc2 = Linear1D_Row.from_native_module(fc2, process_group=tp_group, parallel_input=True)
+
+    def forward(self, x):
+        return self.fc2(self.fc1(x))
+
+
+def set_dist_grad(
+    dist_module: nn.Module, torch_model: nn.Module, g_dtype: torch.dtype, group: dist.ProcessGroup
+) -> None:
+    """
+    Set split grads for Tensor Parallel or ZeRO DP.
+    We do not need a separate treatment for ZeRO,
+    as the wrapper takes care of reduce-scattering grads.
+    """
+    rank = dist.get_rank(group)
+    world_size = dist.get_world_size(group)
+
+    for p, torch_p in zip(dist_module.parameters(), torch_model.parameters()):
+        if torch_p.grad is None:
+            torch_p.grad = torch.zeros_like(torch_p)
+
+        is_distributed = hasattr(p, "dist_layout")
+        if is_distributed:
+            sharding = p.dist_layout.sharding_spec.sharding_sequence
+            split_dim = sharding.index(_TP_SPEC)
+            shape = torch_p.split(world_size, dim=split_dim)[rank].shape
+
+            indices = torch.arange(shape[split_dim] * rank, shape[split_dim] * (rank + 1))
+            # Generate grads only for the correctly split chunk
+            torch_p.grad.index_add_(split_dim, indices, torch.randn(shape, device=torch_p.device, dtype=g_dtype))
+
+        else:
+            shape = torch_p.shape
+            torch_p.grad += torch.randn(shape, device=torch_p.device, dtype=g_dtype)
+
+        # avoid inconsistent grad and param dtype error
+        orig_p = p.data
+        p.data = torch_p.grad.clone().to(g_dtype)
+        p.grad = p.data
+        p.data = orig_p
+
+
+@parameterize("p_g_dtype", _ALLOWED_P_G_TYPES)
+@parameterize("bias_correction", [False, True])
+@parameterize("tp_zero_size", [(4, 1), (2, 2), (1, 4)])
+def run_dist_lamb_optim(
+    bias_correction: bool, p_g_dtype: tuple[torch.dtype, torch.dtype], tp_zero_size: tuple[int, int]
+) -> None:
+    p_dtype, g_dtype = p_g_dtype
+    tp_size, zero_size = tp_zero_size
+    device_mesh, tp_group, dp_group = DistributedLamb.setup_distributed(tp_size, zero_size)
+    rank = dist.get_rank()
+
+    torch_model = Net().to(rank)
+    model = TPNet(torch_model.fc1, torch_model.fc2, tp_group).to(rank)
+
+    # Set up optimizers
+    lr = 1e-3
+    beta1, beta2 = 0.9, 0.999
+    eps = 1e-8
+    torch_optim = Lamb(setup_param_groups(torch_model), lr=lr, betas=(beta1, beta2), eps=eps)
+    optim = DistributedLamb(
+        setup_param_groups(model),
+        lr=lr,
+        betas=(beta1, beta2),
+        eps=eps,
+        bias_correction=bias_correction,
+        device_mesh=device_mesh,
+    )
+    if zero_size > 1:
+        optim = LowLevelZeroOptimizer(
+            optim,
+            overlap_communication=True,
+            initial_scale=128,
+            partition_grad=True,
+            dp_process_group=dp_group,
+            verbose=True,
+        )
+
```

Comment:
Can we generate random grads rather than running forward-backward? E.g. 
```python
for group in optim.param_groups:
    for p in group["params"]:
        p.grad = torch.rand_like(p)
```

Reply:
I have that function above but we need to call zero_optim.backward?
`        if zero_size > 1:
            optim.backward(out_tp)
        else:
            out_tp.backward()
`

Reply:
Don't wrap optimizer with LowLevelZeroOptimizer at this stage. We need 2 unit tests. The first one only tests optimizer itself. And the second one tests the whole training step with Booster API.